src/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCIntegrationTest.java

```diff
@@ -73,6 +73,29 @@ public static void afterClass() throws Exception {
         connection = null;
     }
 
+    @Test
+    public void testConnectionWithCustomSchema() throws Exception {
+        Connection fooConn = DriverManager.getConnection(String.format("crate://%s/foo", hostAndPort));
+        assertThat(fooConn.getSchema(), is("foo"));
+        Statement statement = fooConn.createStatement();
+        statement.execute("create table t (x string) with (number_of_replicas = 0)");
+        statement.execute("insert into t (x) values ('a')");
+        statement.execute("refresh table t");
+        ResultSet resultSet = statement.executeQuery("select count(*) from t");
+        resultSet.next();
+        assertThat(resultSet.getLong(1), is(1L));
+
+        Connection barConnection = DriverManager.getConnection(String.format("crate://%s/bar", hostAndPort));
+        assertThat(barConnection.getSchema(), is("bar"));
+        statement = barConnection.createStatement();
+        statement.execute("create table t (x string) with (number_of_replicas = 0)");
+        statement.execute("insert into t (x) values ('a')");
+        statement.execute("refresh table t");
+        resultSet = statement.executeQuery("select count(*) from t");
+        resultSet.next();
+        assertThat(resultSet.getLong(1), is(1L));
```

Comment:
maybe ensure with a call to information_schema.tables that both tables exist, and no doc.t has been created